controller/workitemtype.go

```diff
@@ -135,9 +135,9 @@ func (c *WorkitemtypeController) List(ctx *app.ListWorkitemtypeContext) error {
 func (c *WorkitemtypeController) ListSourceLinkTypes(ctx *app.ListSourceLinkTypesWorkitemtypeContext) error {
 	return application.Transactional(c.db, func(appl application.Application) error {
 		// Test that work item type exists
-		_, err := appl.WorkItemTypes().Load(ctx.Context, ctx.SpaceID, ctx.WitID)
-		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, err)
+		exists, err := appl.WorkItemTypes().Exists(ctx.Context, ctx.WitID.String())
+		if !exists || err != nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
```

Comment:
I think the check for `err != nil` is in fact enough because you return a `NotFoundError` from `repository.Exists` when the item didn't exist. Please re-check all calls to `Exists`.